File: lib/domain/entities/message.dart
/// Message types
enum MessageType { text, image, video, audio, file }

/// Message delivery status
enum MessageStatus { sending, sent, delivered, read, failed }

/// Auto-delete policy options
enum AutoDeletePolicy { 
  oneDay, 
  threeDays, 
  sevenDays, 
  thirtyDays, 
  afterRead, 
  onExit 
}

/// Message entity representing a chat message
class Message {
  final String id;
  final String chatId;
  final String senderId;
  final String content;
  final MessageType type;
  final DateTime timestamp;
  final MessageStatus status;
  final AutoDeletePolicy? autoDelete;

  const Message({
    required this.id,
    required this.chatId,
    required this.senderId,
    required this.content,
    this.type = MessageType.text,
    required this.timestamp,
    this.status = MessageStatus.sending,
    this.autoDelete,
  });

  Message copyWith({
    String? id,
    String? chatId,
    String? senderId,
    String? content,
    MessageType? type,
    DateTime? timestamp,
    MessageStatus? status,
    AutoDeletePolicy? autoDelete,
  }) {
    return Message(
      id: id ?? this.id,
      chatId: chatId ?? this.chatId,
      senderId: senderId ?? this.senderId,
      content: content ?? this.content,
      type: type ?? this.type,
      timestamp: timestamp ?? this.timestamp,
      status: status ?? this.status,
      autoDelete: autoDelete ?? this.autoDelete,
    );
  }
}

File: app/src/main/java/com/pioneer/messenger/domain/model/Message.kt
package com.pioneer.messenger.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class Message(
    val id: String,
    val chatId: String,
    val senderId: String,
    val content: EncryptedContent,
    val timestamp: Long,
    val type: MessageType = MessageType.TEXT,
    val status: MessageStatus = MessageStatus.SENT,
    val replyToId: String? = null,
    val attachments: List<Attachment> = emptyList(),
    val isEdited: Boolean = false,
    val editedAt: Long? = null
)

@Serializable
data class EncryptedContent(
    val ciphertext: String,
    val nonce: String,
    val senderKeyId: String
)

@Serializable
enum class MessageType {
    TEXT, IMAGE, FILE, AUDIO, VOICE, LOCATION, TASK_UPDATE, SYSTEM
}

@Serializable
enum class MessageStatus {
    SENDING, SENT, DELIVERED, READ, FAILED
}

@Serializable
data class Attachment(
    val id: String,
    val type: AttachmentType,
    val name: String,
    val size: Long,
    val mimeType: String,
    val encryptedUrl: String,
    val thumbnailUrl: String? = null
)

@Serializable
enum class AttachmentType {
    IMAGE, VIDEO, AUDIO, DOCUMENT, ARCHIVE
}
